### PR TITLE
Allow setting transparency for website navbar background + Ctrl+Click TurboWarp button for new tab

### PR DESCRIPTION
**Pull request by iqnite**
### Changes

The color picker of the Navbar color setting in the *Website dark mode and customizable colors* addon now allows changing the transparency.
Ctrl+Clicking the TurboWarp mode button opens it in a new tab when Replace player is enabled.

### Reason for changes

Transparency: The feature was already in the *Scratch 2.0 -> 3.0* addon. The transparency slider disappeared when the settings were moved to the `dark-www` addon. This had the consequence that people who changed the transparency in the old version could not change or disable it anymore. Another reason is that a transparent navbar looks nice :)
TurboWarp: It is useful to have dynamic functionality and can use both modes at once.

### Tests

The color changes don't cause any trouble. I don't understand why the slider was removed.
The TurboWarp changes work, but there is a bug occurring sometimes which hides the Sprite and Script count. I'm not sure if it's caused by this.

### CODE REVIEW

**Review by Samq64 — APPROVED** [review]
Untested

**Comment by iqnite** [comment]
> Untested

What do you mean?

**Comment by Samq64** [comment]
I mean I didn't switch to this branch and test the changes locally to make sure everything works as expected. I'll do that now though just to be sure.

**Comment by Samq64** [comment]
> The feature was already in the Scratch 2.0 -> 3.0 addon.

I'm pretty sure it wasn't, but I'll have to check.

**Comment by mxmou** [comment]
I think the way the text color for the navigation bar is chosen will have to be changed.

**Comment by iqnite** [comment]
> I think the way the text color for the navigation bar is chosen will have to be changed.

uhm...

**Comment by mxmou** [comment]
> > The feature was already in the Scratch 2.0 -> 3.0 addon.
> 
> I'm pretty sure it wasn't, but I'll have to check.

#2097 made it possible to set any color setting to a transparent color. That wasn't intentional though.

**Comment by iqnite** [comment]
> > > The feature was already in the Scratch 2.0 -> 3.0 addon.
> > 
> > 
> > I'm pretty sure it wasn't, but I'll have to check.
> 
> #2097 made it possible to set any color setting to a transparent color. That wasn't intentional though.

@mxmou I think it should like I said in the original PR comment.

**Comment by WorldLanguages** [comment]
> This had the consequence that people who changed the transparency in the old version could not change or disable it anymore.

So was the transparency intentionally changeable at that point, or was it an accidental bug?

**Comment by Samq64** [comment]
> #2097 made it possible to set any color setting to a transparent color. That wasn't intentional though.

**Comment by mxmou** [comment]
Please don't make unrelated changes in the same PR - it makes them more difficult to review. It's better to make a new branch and open a PR from that.

**Comment by iqnite** [comment]
> Please don't make unrelated changes in the same PR - it makes them more difficult to review. It's better to make a new branch and open a PR from that.

Sorry, anyways it's too late. And honestly, 2 days to review 1 added line of code is too much.

**Comment by WorldLanguages** [comment]
> And honestly, 2 days to review 1 added line of code is too much.

Is it "one added line of code", or are you restoring behavior that was never intentional and was simply caused by a bug? If it was accidental, then you can't argue that we should be merging this quickly, because we are not restoring a functionality that was previously possible. This would be, instead, a new feature, and those often take more than 2 days to consider.

**Comment by WorldLanguages** [comment]
This should be split into 2 different PRs. A great opportunity to start playing with git branches.

**Comment by DNin01** [comment]
As WorldLanguages said in the comment above, it would be a good idea to keep each new feature separate by making a pull request just for it. It lets us review and discuss them separately, and also separates them into their own commits which is always a good idea for any version control system.

Unfortunately, the same account can only have one fork of each repository (at least, that's what happened to me when I tried to make a second fork). But fortunately, you can get around this by creating branches.

You can make a new branch by following these steps:
1. Navigate to the repository you want to create a branch for (this will be [your fork](https://github.com/iqnite/ScratchAddons)).
2. Click on Branches
3. Click the "New branch" button
4. Type in the name of your branch (name it something concise and related to your feature, for example, `transparent-navbar`)
5. Choose ScratchAddons/ScratchAddons:master as the branch source. This will copy the contents of the upstream repository (this one) into the branch you are creating. Don't worry if you don't know what that means now, just make sure you do that step.
6. Click "Create branch"

This will also keep your branches up to date which makes it less likely that you'll get merge conflicts.

Feel free to make new pull requests for each feature. You can read our [contributing guidelines](https://github.com/ScratchAddons/ScratchAddons/blob/master/.github/CONTRIBUTING.md), too.

**Comment by iqnite** [comment]
> As WorldLanguages said in the comment above, it would be a good idea to keep each new feature separate by making a pull request just for it. It lets us review and discuss them separately, and also separates them into their own commits which is always a good idea for any version control system.
> 
> Unfortunately, the same account can only have one fork of each repository (at least, that's what happened to me when I tried to make a second fork). But fortunately, you can get around this by creating branches.
> 
> You can make a new branch by following these steps:
> 
> 1. Navigate to the repository you want to create a branch for (this will be [your fork](https://github.com/iqnite/ScratchAddons)).
> 2. Click on Branches
> 3. Click the "New branch" button
> 4. Type in the name of your branch (name it something concise and related to your feature, for example, `transparent-navbar`)
> 5. Choose ScratchAddons/ScratchAddons:master as the branch source. This will copy the contents of the upstream repository (this one) into the branch you are creating. Don't worry if you don't know what that means now, just make sure you do that step.
> 6. Click "Create branch"
> 
> This will also keep your branches up to date which makes it less likely that you'll get merge conflicts.
> 
> Feel free to make new pull requests for each feature. You can read our [contributing guidelines](https://github.com/ScratchAddons/ScratchAddons/blob/master/.github/CONTRIBUTING.md), too.

@DNin01 @WorldLanguages I know, but I don't understand branches. I tried to do that once and it ended up in real chaos with me having 2 separate versions of the same repo with different features. Is there a way to use both branches at once?

File: addons/turbowarp-player/userscript.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ export default async function ({ addon, console, msg }) {
     button.title = "TurboWarp";
   }
 
-  button.onclick = async () => {
```
Comment (GrahamSH-LLK) [inline addons/turbowarp-player/userscript.js:29]: Maybe I'm dumb (and I know you didn't write this code), but shouldn't this be an event listener? I'm pretty sure the standard is to not use the old .ons.

**Comment by WorldLanguages** [comment]
> I tried to do that once and it ended up in real chaos with me having 2 separate versions of the same repo with different features.

@iqnite That's what we're looking for here :) So go ahead with the chaos

File: addons/turbowarp-player/userscript.js (hunk 1 above)
Comment (WorldLanguages) [reply addons/turbowarp-player/userscript.js:29]: It's okay if the event is being registered into an element that was created by the same addon. The new userscript docs will make this explicit

**Comment by DNin01** [comment]
> > I tried to do that once and it ended up in real chaos with me having 2 separate versions of the same repo with different features.
> 
> @iqnite That's what we're looking for here :) So go ahead with the chaos

Yeah, that's basically what branches are supposed to do (and you'd eventually merge them together if you were working on a change). I'd suggest giving it a try. Just make sure you follow the steps I provided. The purpose of branches is a little different for contributing using a fork.

**Comment by iqnite** [comment]
Is it OK if I close this PR and open 2 new ones?

**Comment by DNin01** [comment]
> Is it OK if I close this PR and open 2 new ones?

Sure.

**Comment by iqnite** [comment]
I'm closing this pull request and splitting it into 2 new ones.

**Pull request closed without merging**